ebos/eclgenericvanguard.hh

```diff
@@ -101,13 +114,7 @@ public:
     /*!
      * \brief Set the simulation configuration objects.
      */
-    static void setParams(double setupTime,
-                          std::shared_ptr<EclipseState> eclState,
-                          std::shared_ptr<Schedule> schedule,
-                          std::unique_ptr<UDQState> udqState,
-                          std::unique_ptr<Action::State> actionState,
-                          std::unique_ptr<WellTestState> wtestState,
-                          std::shared_ptr<SummaryConfig> summaryConfig);
+    void setParams(SetupParams& params);
```

Comment:
Given the implementation of this function I don't feel comfortable with a reference-to-mutable parameter.  The opportunities for misuse of the proposed interface are too great in my opinion.  If it were me I would make the parameter into an r-value reference instead and have callers write
```C++
setParams(std::move(some_object))
```
possibly followed by
```C++
some_object.reset() // or something to that effect
```
or
```C++
setParams(construct_parameter_object(from, a, set, of, arguments))
```
instead.

On a tangential note, since it was added earlier and not technically part of this change, I think the `setParams()` name is a little too generic for what this function does.  `AssignInitialRunState()` or, maybe, `defineSimulationModel()` would be more descriptive.

Reply:
okay, let's address the setParams separately first.

Reply:
i made it an rref. however a reset is pointless, we're moving everything out of the struct, so there is nothing to reset (all ptrs will be nullptr's when done).